### DISCUSSION TITLE
Device Name not shown in Tasmota 12.1.0.1 (Dev)

### DISCUSSION THREAD

**Discussion by Mark-Muc** [arendst/Tasmota#16324] [Q&A] [answered]
It seems that this version don't show the Device Name, provided by the template don't show in the WebUI 
i.e.
`{"NAME":"Shelly 2.5 (Taster)","GPIO":[320,0,0,0,224,129,0,0,640,128,608,225,3456,4736],"FLAG":0,"BASE":18}`

Shelly 2.5 (Taster) won't show  in the UI

bevor:
![image](https://user-images.githubusercontent.com/67410847/186351075-5c0fc2b8-ff93-4341-bbd6-3f8b9870009e.png)

after:
![image](https://user-images.githubusercontent.com/67410847/186351743-933de686-3cd5-40d0-879e-2f31da85c73f.png)

Version:
<html>
<body>

Tasmota Version | 12.1.0.1(TasmoCompiler-esp8266generic)
-- | --
2022.08.22 07:31:51
2_7_4_9/2.2.2-dev(38a443e)
0T00:02:09
21 at 0xFB000
10
Software/System restart
Rollo Vorraum Ostauf
Rollo Vorraum Ost zu

</body>
</html>

**Comment by arendst** [comment, marked as answer]
No bug. It's a feature. Read the release notes and find new command ``SetOption141``

**Reply by Mark-Muc** [reply]
oh sorry i missed that .... will have a look . many thx!!
Mark

**Reply by Mark-Muc** [reply]
ok I tried to bring it back with:
`SetOption141 0`
didn't work by me, also with restart etc. Browser in privat mode ..
Mark

**Reply by barbudor** [reply]
I confirm @arendst : On latest build 12.1.0.1 `SO141` do not work as expected
works on 12.0.2.2

**Reply by arendst** [reply]
I'll chk what went wrong.

**Reply by arendst** [reply]
Sorry guys. I cannot reproduce. v12.1.0 works fine. Current dev v12.1.0.1 works fine too.

@barbudor what exact version (binary) doesn't work for you?

**Reply by sfromis** [reply]
I'm not seeing the module name in the header anymore, with 
```
16:22:04.728 CMD: SetOption141
16:22:04.732 MQT: xxxxxxxx/stat/SETOPTION = {"SetOption141":"OFF"}
16:22:09.491 CMD: Module
16:22:09.495 MQT: xxxxxxxx/stat/MODULE = {"Module":{"1":"ESP32C3"}}
```
I'm only seeing the Devicename, or with SetOption53 also hostname/IP. Standard precompiled ESP32-C3 dev build from 4 days ago
`"Version":"12.1.0.1(tasmota)","BuildDateTime":"2022-08-20T11:48:23"`

**Reply by arendst** [reply]
I still cannot reproduce. The code snippet is very simple and always works on my code, the released verseion and latest dev version:

```
const char HTTP_HEAD_STYLE3[] PROGMEM =
  "</style>"

  "</head>"
  "<body>"
  "<div style='text-align:left;display:inline-block;color:#%06x;min-width:340px;'>"  // COLOR_TEXT
#ifdef FIRMWARE_MINIMAL
#ifdef FIRMWARE_SAFEBOOT
  "<span style='text-align:center;color:#%06x;'><h3>" D_SAFEBOOT "</h3></span>"  // COLOR_TEXT_WARNING
#else
  "<div style='text-align:center;color:#%06x;'><h3>" D_MINIMAL_FIRMWARE_PLEASE_UPGRADE "</h3></div>"  // COLOR_TEXT_WARNING
#endif  // FIRMWARE_SAFEBOOT
#endif  // FIRMWARE_MINIMAL
  "<div style='text-align:center;color:#%06x;'><noscript>" D_NOSCRIPT "<br></noscript>" // COLOR_TITLE
/*
#ifdef LANGUAGE_MODULE_NAME
  "<h3>" D_MODULE " %s</h3>"
#else
  "<h3>%s " D_MODULE "</h3>"
#endif
*/
  "<h3>%s</h3>"    // Module name
  "<h2>%s</h2>";   // Device name

  WSContentSend_P(HTTP_HEAD_STYLE3, WebColor(COL_TEXT),
#ifdef FIRMWARE_MINIMAL
  WebColor(COL_TEXT_WARNING),
#endif
  WebColor(COL_TITLE),
  (Web.initial_config) ? "" : (Settings->flag5.gui_module_name) ? "" : ModuleName().c_str(), SettingsText(SET_DEVICENAME));
```

**Reply by arendst** [reply]
GOT IT. The test for initial_config fails when only one SSID is configured. 

As a workaround if you have one SSID not configured, configure it the same as the other. This should fix the initial_config flag.

I'll see how to fix this.

**Reply by arendst** [reply]
Fixed in latest dev v12.1.0.1 - 20220824

**Reply by barbudor** [reply]
I was running a 12.1.0.1 of yesterday evening but you found out in the mean time 😄

and the workaround works fine too

**Reply by Mark-Muc** [reply]
the workaround works for me too .. later i'll try the "newer" built
Thx for finding the problem
Mark

**Reply by Mark-Muc** [reply]
Ver. 12.1.1.1 DOES the trick .. works correct again!!
Mark
